src/skeval/classifier/sentence_classifier.py

```
@@ -458,8 +458,13 @@ def load(self, save_dir: str) -> None:
         Args:
             save_dir: Directory that was passed to a previous ``save()`` call.
         """
-        with open(os.path.join(save_dir, "metadata.json"), "r") as f:
-            meta = json.load(f)
+        try:
+            with open(os.path.join(save_dir, "metadata.json"), "r") as f:
+                meta = json.load(f)
+        except FileNotFoundError:
+            raise FileNotFoundError(
+                f"No saved model found in '{save_dir}'. Call save() first."
+            )
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)